packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -773,54 +774,64 @@ class _ModalScopeState<T> extends State<_ModalScope<T>> {
 
   @override
   Widget build(BuildContext context) {
-    return _ModalScopeStatus(
-      route: widget.route,
-      isCurrent: widget.route.isCurrent, // _routeSetState is called if this updates
-      canPop: widget.route.canPop, // _routeSetState is called if this updates
-      child: Offstage(
-        offstage: widget.route.offstage, // _routeSetState is called if this updates
-        child: PageStorage(
-          bucket: widget.route._storageBucket, // immutable
-          child: Actions(
-            actions: _actionMap,
-            child: FocusScope(
-              node: focusScopeNode, // immutable
-              child: RepaintBoundary(
-                child: AnimatedBuilder(
-                  animation: _listenable, // immutable
-                  builder: (BuildContext context, Widget? child) {
-                    return widget.route.buildTransitions(
-                      context,
-                      widget.route.animation!,
-                      widget.route.secondaryAnimation!,
-                      // This additional AnimatedBuilder is include because if the
-                      // value of the userGestureInProgressNotifier changes, it's
-                      // only necessary to rebuild the IgnorePointer widget and set
-                      // the focus node's ability to focus.
-                      AnimatedBuilder(
-                        animation: widget.route.navigator?.userGestureInProgressNotifier ?? ValueNotifier<bool>(false),
-                        builder: (BuildContext context, Widget? child) {
-                          final bool ignoreEvents = _shouldIgnoreFocusRequest;
-                          focusScopeNode.canRequestFocus = !ignoreEvents;
-                          return IgnorePointer(
-                            ignoring: ignoreEvents,
-                            child: child,
+    return AnimatedBuilder(
+      animation: widget.route.restorationScopeId,
```

Comment:
why do we use restorationScopeId here?

Reply:
Whenever the restorationScopeID changes, we need to rebuild the RestorationScope.

Reply:
Ah I was confused by the the name animated, and keep thinking it should be a double.